integrationtests/src/test/java/io/kroxylicious/proxy/ProxyRpcTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.apache.kafka.common.message.ApiMessageType;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.kroxylicious.test.ApiMessageSampleGenerator;
+import io.kroxylicious.test.ApiMessageSampleGenerator.ApiAndVersion;
+import io.kroxylicious.test.Request;
+import io.kroxylicious.test.Response;
+import io.kroxylicious.test.client.KafkaClient;
+import io.kroxylicious.test.server.MockServer;
+
+import static io.kroxylicious.proxy.Utils.startProxy;
+import static org.apache.kafka.common.protocol.ApiKeys.API_VERSIONS;
+import static org.apache.kafka.common.protocol.ApiKeys.CONTROLLED_SHUTDOWN;
+import static org.apache.kafka.common.protocol.ApiKeys.SASL_AUTHENTICATE;
+import static org.apache.kafka.common.protocol.ApiKeys.SASL_HANDSHAKE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ProxyRpcTest {
+
+    public static final int PROXY_PORT = 9192;
+    public static final String PROXY_ADDRESS = "localhost:" + PROXY_PORT;
+    private static MockServer mockServer;
+    private static KafkaClient kafkaClient;
+    private static KafkaProxy kafkaProxy;
+
+    public record Scenario(String name, Response givenMockResponse, Request whenSendRequest, Request thenMockReceivesRequest, Response thenResponseReceived) {
+    }
+
+    /**
+     * API_VERSIONS is not proxied, kroxy can respond to this itself
+     */
+    private static final Set<ApiKeys> NOT_PROXIED = Set.of(API_VERSIONS);
+
+    @BeforeAll
+    public static void beforeAll() throws InterruptedException {
+        mockServer = MockServer.startOnRandomPort();
+        kafkaClient = new KafkaClient();
+        String config = configure(mockServer);
+        kafkaProxy = startProxy(config);
```

Comment:
I was expecting to see you start a proxy configured in such a way that it has to decode all the RPCs.

Reply:
Yep, it is:

```
kroxyConfigBuilder.withFilter("FixedClientId", Map.of("clientId", "fixed"));
```

Is a filter which encodes/decodes all and sets the clientId header